Lesson5/src/service/MilitaryOffice.java

```diff
@@ -0,0 +1,87 @@
+package service;
+
+import model.Address;
+import model.Person;
+import model.Recruit;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MilitaryOffice {
+    private PersonRegistry personRegistry;
+    private List<Person> fitForArmy = new ArrayList<>();
+    private List<MilitaryBase> militaryBaseList;
+
+    public MilitaryOffice(PersonRegistry personRegistry, List<MilitaryBase> militaryBase) {
+        this.personRegistry = personRegistry;
+        this.militaryBaseList = militaryBase;
+    }
+
+    public List<Person> getHealthyPeople(Address address) {
+        for (Person person : personRegistry.getPeople(address)) {
+            if (person.getSex().equals("man") &&
+                    person.getAge() > 17 && person.getAge() < 28) {
+                System.out.println(person.getName());
+                fitForArmy.add(person);
+            }
+        }
+        return fitForArmy;
+    }
+
+    public List<Person> getHealthyPeople(String country) {
+        for (Person person : personRegistry.getCitizens()) {
+            if (person.getAddress().getCountry().equals(country) &&
+                    person.getSex().equals("man") &&
+                    person.getAge() > 17 && person.getAge() < 28) {
+                System.out.println(person.getName() + "\n");
+                fitForArmy.add(person);
+            }
+        }
+        return fitForArmy;
+    }
+
+    public int getCountTheNeededPeople() {
+        int countMaxNeededPeople = 0;
+        for (MilitaryBase militaryBase : militaryBaseList) {
+            countMaxNeededPeople += militaryBase.getAvailablePlaces();
+        }
+        return countMaxNeededPeople;
+    }
+
+    public void addFitPeopleToTheUnits(Address address) {
+        List<Person> fitPeople = getHealthyPeople(address);
+        List<Person> peopleForMilitaryBase = new ArrayList<>();
+        if (fitPeople.size() < getCountTheNeededPeople()) {
+            for (MilitaryBase militaryBase : militaryBaseList) {
+                for (int i = 0; i < militaryBase.getAvailablePlaces(); i++) {
+                    if (i < fitPeople.size()) {
+                        peopleForMilitaryBase.add(fitPeople.get(i));
+                    }else{
+                        break;
+                    }
+                }
+            }
+        }else{
+            System.out.println("There are more people than the right places");
+        }
+    }
+
+    public void addFitPeopleToTheUnits(String country) {
+        List<Person> fitPeople = getHealthyPeople(country);
+        List<Person> peopleForMilitaryBase = new ArrayList<>();
+        if (fitPeople.size() < getCountTheNeededPeople()) {
+            for (MilitaryBase militaryBase : militaryBaseList) {
+                for (int i = 0; i < militaryBase.getAvailablePlaces(); i++) {
```

Comment:
imagine this situation

you have 2 military bases, first has 2 free places, second has 1 free place

and you have 3 fitPeople

What will happen here
first for statement will look like

for (int i = 0; I < 2; i++) {
...
peopleForMilitaryBase.add(fitPeople.get(i));
...
}

so you will add first and second fit person to the first military base

second for will look like 
for (int i = 0; I < 1; i++) {
...
peopleForMilitaryBase.add(fitPeople.get(i));
...
}

so you will add  first person (which is already on first military base) to the second military base.
And third person will not be added anywhere. So this code requires fix